src/main/java/com/razorpay/Utils.java

```diff
@@ -0,0 +1,75 @@
+package com.razorpay;
+
+import org.json.JSONObject;
+
+import java.math.BigInteger;
+import java.util.Base64;
+
+import javax.crypto.Mac;
+
+import javax.crypto.spec.SecretKeySpec;
+
+public class Utils {
+
+  protected String key;
+
+  protected String secret;
+
+  public Utils(String key, String secret) {
+    this.key = key;
+    this.secret = secret;
+  }
+
+  public boolean verifyPaymentSignature(JSONObject attributes)
+  {
+    String expectedSignature = attributes.getString("razorpay_signature");
+    String orderId = attributes.getString("razorpay_order_id");
+    String paymentId = attributes.getString("razorpay_payment_id");
+    String payload = orderId + '|' + paymentId;
+
+    return this.verifySignature(payload, expectedSignature);
+  }
+
+  public boolean verifyWebhookSignature(String payload, String expectedSignature)
+  {
+    return this.verifySignature(payload, expectedSignature);
+  }
+
+  public boolean verifySignature(String payload, String expectedSignature)
+  {
+    String actualSignature = this.getHash(payload);
+        
+    return this.isEqual(actualSignature.getBytes(), expectedSignature.getBytes());
```

Comment:
@ankit-rzp I think this is intentional: http://security.stackexchange.com/a/83670